pkg/api/job_runs.go

```diff
@@ -467,6 +467,14 @@ func runJobRunAnalysis(jobRun *models.ProwJobRun, compareRelease string, jobRunT
 		if testResultsJobNameFunc != nil {
 			if len(jobNames) < 5 && len(jobNames) > 0 {
 				testResultsJobNames, errJobNames = testResultsJobNameFunc(ft.Test.Name, jobNames)
+
+				if errJobNames == nil && testResultsJobNames != nil {
+					if testResultsJobNames.CurrentRuns == 0 {
+						// do we need to prepend the suite name to the test?
+						testResultsJobNames, errJobNames = testResultsJobNameFunc(fmt.Sprintf("%s.%s", ft.Suite.Name, ft.Test.Name), jobNames)
```

Comment:
is that a suite name?

There are 3 suites:

```sql
sippy_openshift=> select name from suites ;
          name           
-------------------------
 openshift-tests
 openshift-tests-upgrade
 sippy
(3 rows)
```

But looking for that pattern above, there are a lot of other things that come before the ".""

```sql
sippy_openshift=> SELECT DISTINCT
    SUBSTRING(name FROM 1 FOR POSITION('.' IN name) - 1) AS name_prefix
FROM
    tests
WHERE
    name LIKE '%.[%'
ORDER BY
    name_prefix;
                                name_prefix                                 
----------------------------------------------------------------------------
 Backup from one node and recover on another
 Cluster API Suite
 Cluster Controller Suite
 Cluster upgrade
 CNF Features e2e integration tests
 CNF Features e2e setup
 CNF Features e2e validation
 Controller Suite
 E2E Suite
 End-to-end
 Environment Setup Suites
 github
 Integration Suite
 Kubeconfig Controller Suite
 Kubernetes e2e suite
 Machine Suite
 metal infra
 Metallb Operator E2E Suite
 Metallb Operator Validation Suite
 OADP E2E Suite
 odo e2e scenarios
 OSD e2e suite
 Performance Addon Operator configuration
 Performance Addon Operator latency e2e tests
 Performance Test
 PlatformOperators Suite
 PTP e2e integration tests
 PTP e2e tests : Parallel
 Quorum Loss and Restore
 Repos
 [sig-storage] Ensure that all storage objects are deleted during the tests
 SRIOV Operator conformance tests
 Subsystem Suite
 Tests Suite
 Test Suites
(35 rows)
```

Does `ft.Suite.Name` get you what you want?

Reply:
This is precisely the trouble I got into when trying to split the suite names out, possibly related to some legacy compatability we may no longer care about. When we parse the xml, we should explicitly know the suite name. It's just a question of migrating data perhaps.